full/src/test/java/apoc/uuid/UUIDNewProceduresTest.java

```diff
@@ -0,0 +1,494 @@
+
+        // then
+        try (Transaction tx = db.beginTx()) {
+            Node n = (Node) tx.execute("MATCH (person:Person) return person").next().get("person");
+            assertTrue(n.getAllProperties().containsKey("uuid"));
+            assertIsUUID(n.getAllProperties().get("uuid").toString());
+            tx.commit();
+        }
+    }
+
+    @Test
+    public void testRemoveAllUuid() {
+        // given
+        db.executeTransactionally("CREATE CONSTRAINT ON (test:Test) ASSERT test.foo IS UNIQUE");
+        db.executeTransactionally("CREATE CONSTRAINT ON (bar:Bar) ASSERT bar.uuid IS UNIQUE");
+        sysDb.executeTransactionally("CALL apoc.uuid.create('neo4j', 'Bar') YIELD label RETURN label");
+        sysDb.executeTransactionally("CALL apoc.uuid.create('neo4j', 'Test', {addToExistingNodes: false, uuidProperty: 'foo'}) YIELD label RETURN label");
+        testCallCountEventually(db, "CALL apoc.uuid.list", 2, TIMEOUT);
+
+        // when
+        TestUtil.testResult(sysDb, "CALL apoc.uuid.dropAll('neo4j')",
+                (result) -> {
+                    // then
+                    Map<String, Object> row = result.next();
+                    assertResult(row, "Bar", false,
+                            Util.map(UUID_PROPERTY_KEY, "uuid", ADD_TO_SET_LABELS_KEY, false));
+                    row = result.next();
+                    assertResult(row, "Test", false,
+                            Util.map(UUID_PROPERTY_KEY, "foo", ADD_TO_SET_LABELS_KEY, false));
+                });
+
+        testCallCountEventually(db, "CALL apoc.uuid.list", 0, TIMEOUT);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void testAddWithError() {
+        try {
+            // when
+            sysDb.executeTransactionally("CALL apoc.uuid.create('neo4j', 'Wrong') YIELD label RETURN label");
+        } catch (RuntimeException e) {
+            // then
+            Throwable except = ExceptionUtils.getRootCause(e);
+            assertTrue(except instanceof RuntimeException);
+            assertEquals("No constraint found for label: Wrong, please add the constraint with the following : `CREATE CONSTRAINT ON (wrong:Wrong) ASSERT wrong.uuid IS UNIQUE`", except.getMessage());
+            throw e;
+        }
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void testAddWithErrorAndCustomField() {
+        try {
+            // when
+            sysDb.executeTransactionally("CALL apoc.uuid.create('neo4j', 'Wrong', {uuidProperty: 'foo'}) YIELD label RETURN label");
+        } catch (RuntimeException e) {
+            // then
+            Throwable except = ExceptionUtils.getRootCause(e);
+            assertTrue(except instanceof RuntimeException);
+            assertEquals("No constraint found for label: Wrong, please add the constraint with the following : `CREATE CONSTRAINT ON (wrong:Wrong) ASSERT wrong.foo IS UNIQUE`", except.getMessage());
+            throw e;
+        }
+    }
+
+    //
```

Comment:
should we also have a test that has UUID installed for two labels, and then we create a node with one label and add the other? so it should not override the previous uuid and handle it gracefully without failing.

for that situation we should probably also make sure to only remove the uuid from the node if has no labels that have  a uuid handler installed?

Reply:
Or we need to disallow it altogether, which would be probably hard to enforce.